backend/src/main/kotlin/hu/bme/sch/cmsch/component/app/ApplicationService.kt

```diff
@@ -0,0 +1,38 @@
+package hu.bme.sch.cmsch.component.app
+
+import hu.bme.sch.cmsch.component.login.CmschUser
+import hu.bme.sch.cmsch.model.UserEntity
+import org.springframework.stereotype.Service
+
+@Service
+class ApplicationService {
+
+    fun getUserAuthInfo(jwtUser: CmschUser, actualUser: UserEntity?): UserAuthInfoView {
+        if (actualUser == null || isAuthenticationExpired(jwtUser, actualUser))
+            return UserAuthInfoView(authState = AuthState.EXPIRED)
+
+        return UserAuthInfoView(
+            authState = AuthState.LOGGED_IN,
+            id = actualUser.id,
+            internalId = actualUser.internalId,
+            role = actualUser.role,
+            permissionsAsList = actualUser.permissionsAsList,
+            userName = actualUser.userName,
+            groupId = actualUser.groupId,
+            groupName = actualUser.groupName,
+        )
+    }
+
+    fun isAuthenticationExpired(jwtUser: CmschUser, actualUser: UserEntity): Boolean {
```

Comment:
Should I check additional stuff? this is the only data stored in the jwt @berenteb @Gerviba